switches from React.createClass to extend React.Component

Jascha-Sundaresan opened the pull request:
React.createClass is deprecated.

Commit:
switches from React.createClass to extend React.Component

gilbox commented on src/index.js:
```diff
@@ -13,57 +13,62 @@ function factory(options) {
+    return class SparkScroll extends React.Component {
+      constructor(props) {
+        super(props);
+      }
```
the constructor is unecessary

gilbox commented on src/index.js:
```diff
@@ -13,57 +13,62 @@ function factory(options) {
+    return class SparkProxy extends React.Component {
+      constructor(props) {
+        super(props);
+      }
```
this is also unecesary

gilbox reviewed (changes requested):
thanks! just two minor things

Commit:
upgrades react to 15 and fixes deprecation warnings

Commit:
removes unnecessary constructors

Jascha-Sundaresan commented:
Ok.. I removed the unnecessary constructors.

I also updated everything in package.json except babel and webpack as I didn't want to figure out how to fix the breaking changes.

react 15 threw some warnings about tags getting unknown props, so I fixed that, too.

Commit:
removes pure render mixin and switches demo app to extend React.Component

Jascha-Sundaresan commented:
sweet!

gilbox commented:
thanks for your help! I just published v4.0.0

Jascha-Sundaresan commented:
awesome.  My pleasure.
